### PR TITLE
🚀 additional user info added

### PR DESCRIPTION
**Pull request by ssibrahimbas**
## Pull request checklist

It is a fork created for the #30 issue.

What Has Been Done?

- The changes have been tested successfully.
- Created `AdditionalUserInfo.java` class.
- Edited `SignInResult` interface in `src/definitions.ts`.
- `AdditionalUserInfo` object is integrated in Android for all supporting providers.

### CODE REVIEW

**Comment by robingenz** [comment]
Thank you! I will review it as soon as I have time.

File: packages/authentication/src/definitions.ts [hunk 1]
```diff
@@ -326,6 +331,11 @@ export interface User {
   uid: string;
 }
 
+export interface AdditionalUserInfo {
+  userId?: string | null;
+  email?: string | null;
```
Comment (robingenz) [inline packages/authentication/src/definitions.ts:336]:
```suggestion
  isNewUser: boolean;
  username?: string;
  profile?: { [key: string]: any; };
```

It's best to follow the official JS SDK for this interface: https://firebase.google.com/docs/reference/js/auth.additionaluserinfo#properties

**Review by robingenz — CHANGES_REQUESTED** [review]
Am I missing something or did you forget to commit the `AdditionalUserInfo` file?
Please adapt your implementation to the new interface (see my comments below).

Please also add an implementation for web and iOS.

**Comment by ssibrahimbas** [comment]
All right, I want to you to give me a little time. I Will take the following steps at a convenient time:

1. Update current system according to Firebase Documentation
2. Provide support for IOS and Web
3. Test step 1 and 2
4. Update pull request

I'II let you know!

**Comment by robingenz** [comment]
Closed as inactive.

**Pull request closed without merging**